packages/app/src/signIn/SignInScreen.tsx

```diff
@@ -95,11 +104,29 @@ export default function SignInScreen() {
   }, [authenticationClient, overrideEmailAddress]);
 
   const userRecord = useCurrentUserRecord(authenticationClient);
+
+  React.useEffect(() => {
+    if (userRecord && shouldDoubleCheckAccountStatus === null) {
+      // a user record was received before the user entered their credentials
+      // lets double check they want to use this account
+      setShouldDoubleCheckAccountStatus("requiresUserApproval");
+    }
+  }, [userRecord, shouldDoubleCheckAccountStatus]);
+
+  React.useEffect(() => {
+    if (userRecord && userRecord.emailAddress !== overrideEmailAddress) {
+      authenticationClient.signOut();
+      setShouldDoubleCheckAccountStatus(null);
+    }
+  }, [authenticationClient, userRecord, overrideEmailAddress]);
```

Comment:
Ditto this one: maybe best to make one big effect which amounts to "we got a new user state (or account approval); let's see what needs doing." Wish I could see a better way to simplify this. I suppose we could reify the state machine more explicitly, but I'm not sure it's worth it.